File: Leetcode/905/Selection_Sort_Algo/main.kt
class Solution {
    fun sortArrayByParity(nums: IntArray): IntArray {
        val n = nums.count()
        for (i in 0 until n) {
            for (j in i + 1 until n) {
                if (nums[j] % 2 == 0) {
                    // swap
                    val temp = nums[i]
                    nums[i] = nums[j]
                    nums[j] = temp
                    break
                }
            }
        }
        return nums
    }
}

File: Leetcode/905/Selection_Sort_Algo/main.swift
class Solution {
    func sortArrayByParity(_ nums: [Int]) -> [Int] {
        var nums = nums
        var n: Int = nums.count
        for i in 0..<n { 
            for j in (i+1)..<n {
                if nums[j] % 2 == 0 {
                    // swap
                    var temp: Int = nums[i]
                    nums[i] = nums[j]
                    nums[j] = temp
                    break
                }
            }
        }
        return nums
    }
}